src/main/java/com/epam/test_generator/dao/impl/CaseVersionDAOImpl.java

```diff
@@ -66,4 +92,106 @@ public void delete(Iterable<Case> cases) {
             delete(caze);
         }
     }
+
+    private List<CaseVersion> getCaseVersions(List<Change> changes) {
+        TreeMap<CommitMetadata, List<Change>> changesWithSameCommitId =
+            getSortedGroupedMapByChangesWithSameCommitId(changes);
+
+        List<CaseVersion> caseVersions = new ArrayList<>(changesWithSameCommitId.size());
+        changesWithSameCommitId.forEach(
+            (commitMetadata, changeList) -> {
+                CaseVersion caseVersion = new CaseVersion();
+
+                caseVersion.setAuthor(commitMetadata.getAuthor());
+                caseVersion.setCommitId(commitMetadata.getId().toString());
+                caseVersion.setUpdatedDate(Date.from(commitMetadata.getCommitDate().atZone(
```

Comment:
what about custom constructor for caseVersion?

Reply:
done